aten/src/ATen/native/native_functions.yaml

```diff
@@ -50,6 +50,14 @@
 - func: align_tensors(Tensor[] tensors) -> Tensor[]
   named_guard: False
 
+- func: unflatten(Tensor self, Dimname dim, int[] sizes, DimnameList names) -> Tensor
```

Comment:
is there any close equivalent in numpy?

Reply:
No, in numpy one would use `reshape`. I did some research into the naming of this a while back, below is a summary:

In xarray (which is numpy + named axis), they use [`stack` and `unstack`](http://xarray.pydata.org/en/stable/reshaping.html#stack-and-unstack). The naming is based on pandas arrays' `stack` and `unstack`, which work in a similar manner.

In addition, Sasha Rush's namedtensor design uses [`stack` and `split`](https://github.com/harvardnlp/namedtensor/blob/master/namedtensor/core.py#L112-L120). 

Finally, salesforce/matchbox (autobatching with pytorch tensors) uses `split_dim`.

My proposal is to use `flatten` and `unflatten` instead of the above names. This is because `stack` and `split` already have semantic meaning: in pytorch, stack is an operation that takes a list of tensors and returns a new tensor; split is an operation that takes a tensor and splits it into multiple other tensors. Something like `stack_dim` and `split_dim` would make the semantics clearer, but since we already have `flatten` as a part of the API, `unflatten` is a reasonable counterpart.

Reply:
@zou3519 actually this probably also needs an entry in the docs

Reply:
No named tensors docs yet, but they'll come before the release.